### PR TITLE
Improvements to the header

### LINKED ISSUE #557: Improvements to the on-chain balance overlay
There are several improvements I would like to see on the on-chain balance overlay:

1. The currently selected token should come first in the list.
2. ~The on-chain balance (not deposited to UDC) of the services token (RDN) should also be in the list.~
3. The token mint button should be moved to the on-chain balance overlay. It is currently confusing that it shows up next to the off-chain balance as it does not affect it.

### PR DESCRIPTION
**Pull request by manuelwedler**
Fixes #557

This PR prepares the header for adding the UDC deposit and withdraw functionality (https://github.com/raiden-network/webui/issues/556) according to Sash's updated design. 

We moved the mint button next to the on-chain balance. Before it was next to the off-chain balance even though it doesn't affect it.
Having the selected token first in the header is necessary for being able to still mint all tokens.

### CODE REVIEW

File: src/app/components/header/header.component.html [hunk 1]
```diff
@@ -100,21 +105,36 @@
...
                     <div class="divider"></div>
                     <button
-                        mat-flat-button
+                        mat-icon-button
                         class="balances-button"
+                        matTooltip="{{
+                            tokenBalancesOpen
+                                ? ''
+                                : 'Show all on-chain balances'
+                        }}"
```
Comment (weilbith) [inline src/app/components/header/header.component.html:131]: Nitpick: Sorry I forgot again if this was easily possible in Angular. But this really bloats the template with details. If possible it would be cool if this could be defined as some kind of computed data and you just have something like `matTooltip="{{ showBalancesTooltip }}"`. What do you think?

File: src/app/components/header/header.component.scss [hunk 2]
```diff
@@ -2,22 +2,25 @@
...
 
     &--mobile {
         height: unset;
     }
 
     &__balance {
-        text-transform: uppercase;
+        color: $light-grey;
+
+        &--eth {
```
Comment (weilbith) [inline src/app/components/header/header.component.scss:15]: Nitpick: modifiers of the BEM scheme are usually behavior descriptive adjectives. So eventually `--upper-case` or similar fits better. This would also make it re-usable in case you would need it for anything else that needs the same modification. But this is really a stupid mistake. :laughing:

File: src/app/components/header/header.component.html [hunk 3]
```diff
@@ -81,28 +81,45 @@
...
-                            [token]="token"
-                        ></app-balance-with-symbol>
+                        <span fxLayout="row" fxLayoutGap="6px">
+                            <app-balance-with-symbol
+                                class="header__balance"
+                                [balance]="token.balance"
+                                [token]="token"
+                            ></app-balance-with-symbol>
+                            <button
+                                *ngIf="(network$ | async)?.chainId !== 1"
```
Comment (weilbith) [inline src/app/components/header/header.component.html:98]: I would consider to give this expression a helpful name that reads better. E.g. `ngIf="!connectedToMainnet"` What do you think?

File: src/app/components/header/header.component.ts [hunk 4]
```diff
@@ -136,4 +138,22 @@ export class HeaderComponent implements OnInit, OnDestroy {
     tokenTrackByFn(token: UserToken): string {
         return token.address;
     }
+
+    mint(token: UserToken) {
+        this.mintPending[token.address] = true;
+        const decimals = token.decimals;
+        const scaleFactor = decimals >= 18 ? 1 : decimals / 18;
```
Comment (weilbith) [inline src/app/components/header/header.component.ts:145]: The test cases describe that you care about tokens with 18 decimals or 0. Are those only the cases of interest and the rest just happens but is fine? So when my imaginary token has 9 decimals, this would lead to a `scaleFactor` of 0.5. Is this intended? I just wanna make this sure.

**Review by weilbith — APPROVED** [review]
Looks pretty good to me. :+1: 
Just minor things and some questions. :smiley:

File: src/app/components/header/header.component.html (hunk 3 above)
Comment (manuelwedler) [reply src/app/components/header/header.component.html:98]: Yes, makes sense as it also was like this before :)

File: src/app/components/header/header.component.scss (hunk 2 above)
Comment (manuelwedler) [reply src/app/components/header/header.component.scss:15]: You are totally right :)

File: src/app/components/header/header.component.ts (hunk 4 above)
Comment (manuelwedler) [reply src/app/components/header/header.component.ts:145]: Yes, that's what it should do. 
There are just tests for the boundaries. I think we shouldn't overtest this as it is just a testnet functionality.

Comment (weilbith) [reply src/app/components/header/header.component.ts:145]: Perfect. If that is the case I agree there that is no need to test all boundary groups etc. :heavy_check_mark:

File: src/app/components/header/header.component.html (hunk 1 above)
Comment (manuelwedler) [reply src/app/components/header/header.component.html:131]: There's not a feature like the computed data of Vue. You can use a function but that would get invoked for every change detection. But I just found out that there is `matTooltipDisabled` property you can use to not show the tooltip conditionally. That makes more sense in this case. Thank you for helping me finding this!
